speech-to-text/STT_Inference_API_Server_Colab_Notebook.ipynb

```diff
@@ -0,0 +1,249 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "4MIeAXHiLjLm"
+      },
+      "outputs": [],
+      "source": [
+        "# Import libraries\n",
+        "import uvicorn\n",
+        "from fastapi import FastAPI\n",
+        "from fastapi.middleware.cors import CORSMiddleware\n",
+        "from pyngrok import ngrok\n",
+        "from pydantic import BaseModel\n",
+        "import nest_asyncio\n",
+        "\n",
+        "import shutil\n",
+        "\n",
+        "nest_asyncio.apply()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "metadata": {
+        "id": "DGpm5GvgL4Qr"
+      },
+      "outputs": [],
+      "source": [
+        "# Import models for serialisation/ deserialisation\n",
+        "from pydantic import BaseModel\n",
+        "import base64\n",
+        "import io\n",
+        "import wave\n",
+        "\n",
+        "\n",
+        "class AudioBytes(BaseModel):\n",
+        "  data: bytes\n",
+        "  filename: str\n",
+        "\n",
+        "\n",
+        "app = FastAPI()\n",
+        "\n",
+        "app.add_middleware(\n",
+        "    CORSMiddleware,\n",
+        "    allow_origins=['*'],\n",
+        "    allow_credentials=True,\n",
+        "    allow_methods=['*'],\n",
+        "    allow_headers=['*'],\n",
+        ")\n",
+        "\n",
+        "@app.get(\"/get-transcript\")\n",
+        "async def get_transcript(audio: AudioBytes):\n",
+        "  decoded_data = base64.b64decode(audio.data)\n",
+        "\n",
+        "  # Write bytes data to a .wav file\n",
+        "  with io.BytesIO(decoded_data) as audio_file:\n",
+        "    with wave.open(audio_file, \"wb\") as wav:\n",
+        "      wav.setnchannels(1)\n",
+        "      wav.setsampwidth(2)\n",
+        "      wav.setframerate(16000)\n",
+        "\n",
+        "      # Write .wav files\n",
+        "      wav.writeframes(decoded_data)\n",
+        "\n",
+        "  # Save the audio file with the custom name\n",
+        "    audio_filename = audio.filename\n",
+        "    with open(audio_filename, \"wb\") as file:\n",
+        "        file.write(decoded_data)\n",
+        "\n",
+        "   # Get transcript and delete temporary audio file\n",
+        "  print(\"audio_filename \",audio_filename)\n",
+        "  transcript = transcribe(audio_filename)\n",
+        "  os.remove(audio_filename)\n",
+        "\n",
+        "  return {\"transcript\": transcript}\n",
+        "\n",
+        "@app.get(\"/\")\n",
+        "async def home():\n",
+        "  return {\"msg\":\"Hello from STT.\"}\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 7,
+      "metadata": {
+        "id": "Rrh5RexfIrg7"
+      },
+      "outputs": [],
+      "source": [
+        "# Attach personal token\n",
+        "!ngrok config add-authtoken 2SAgVMxoWLLZzs3vhdokV7YpMKz_3cX2qcBiNSc2CjMKSRqh5"
```

Comment:
I am wondering if we should remove the authtoken here since it was created from our account. typically the guidance is to remove tokens and api keys from the repo. So would say we should remove this for security reasons

Reply:
That's true. I should probably change it to something like "<place_your_access_token here>".

Reply:
yeah. that should work. you can use "<place_your_ngrok_access_token_here>"

Reply:
Sounds good.